library/HelloSign/TemplateSignatureRequest.php

```diff
@@ -120,11 +120,11 @@ public function setCC($role, $email)
      * @param  string $required   (true to allow Editable Merge, only supported for single signer requests)
      * @return TemplateSignatureRequest
      */
-    public function setCustomFieldValue($field_name, $value, $editor = "", $required = false)
+    public function setCustomFieldValue($field_name, $value, $editor = null, $required = false)
     {
         if(empty($this->custom_fields)) $this->custom_fields = json_encode(array());
         $array = json_decode($this->custom_fields, true);
-        if($required) {
+        if(!empty($editor) && $required) {
```

Comment:
This should actually be `if(!empty($editor))`.

If a user passes in `editor => "David", requited => false` then it would not assign the field as editable for David. _Required_ denotes if it is required for the signer to fill in the data. It is an optional parameter independent of _editor_, but can only be used when _editor_ is present.